lib/ansible/module_utils/basic.py

```diff
@@ -1722,7 +1722,7 @@ def atomic_move(self, src, dest, unsafe_writes=False):
                     #     file = _os.path.join(dir, pre + name + suf)
                     # TypeError: can't concat bytes to str
                     error_msg = ('Failed creating tmp file for atomic move.  This usually happens when using Python3 less than Python3.5. '
-                                 'Please use Python2.x or Python3.5 or greater.')
+                                 'Please use Python2.x or Python3.6 or greater.')
```

Comment:
Since we're already enforcing the use of Python 2.7 or 3.6+ at the top of the file, it looks like we can remove the entire `TypeError` handler, rather than just updating the error message. In fact, we should have been able to do that quite some time ago, since we've required Python 2.x or 3.5+ since at least 2.12.

Reply:
If you want to make the exception handler change in a separate PR (along with the `JSONDecodeError` change), that's fine too. Just let me know which way you'd like to handle it.

Reply:
The "usually" in this error message, and the fact we've never supported Python < 3.5 made me hesitate to remove it. I think you're right, that it could have been removed back in 2.5 when we added the check at the top of the file. We didn't enforce the version for 2.3 which is when this was added. Want me to put the JSONDecodeError change back if I remove this section?